docs/data-sources/dashboards.md

```diff
@@ -0,0 +1,43 @@
+---
+subcategory: "Workspace"
+---
+# databricks_dashboards Data Source
+
+This data source allows you to retrieve information about Databricks [Dashboards](https://docs.databricks.com/en/dashboards/index.html).
+
+## Example Usage
+
+```hcl
+data "databricks_dashboards" "all" {
+}
+
+resource "databricks_permissions" "dashboards_permissions" {
+  depends = [ data.databricks_dashboards.all ]
+  for_each = data.databricks_dashboards.this.dashboards[*].dashboard_id
+
+  dashboard_id = each.value
+
+  access_control {
+    group_name       = "Example Group"
+    permission_level = "CAN_MANAGE"
+  }
+}
+
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name_contains` - (Optional) A **case-insensitive** substring to filter Dashboards by their display name.
```

Comment:
Do we really need to support this in the provider? Since we're doing client-side filtering, customers can also implement this filtering as:
```hcl
locals {
    filtered_dashboards = [for dash in data.databricks_dashboards.all.dashboards: dash if strcontains(dash.display_name, "...")]
}
```

Reply:
this is for consistency with other data sources where filtering by name is supported. our other data sources also does client-side filtering, like `clusters` or `jobs`